### ISSUE TITLE
add newsitem and send tweet for recent releases

### ISSUE THREAD

**Issue by robertoostenveld** [fieldtrip/website#281]
to follow up on #275 let's give it a try...

Dear tweetmeisters @Spaak and @elshafeh, can you please handle #277, #278, #279 and #208? Since there are 2 releases out of these 4 that were judged newsworthy, I propose that both of you take one that is to be posted and one that is to be dropped. 

Please do your edits (or close the PR without merging to drop it). The `tweet` section in the heading will be tweeted, the whole news item will appear on the homepage.

**Comment by elshafeh** [comment]
This evening I'll announce that we're changing our tweeting system and tomorrow will tweet about the new releases. And I imagine you meant #280 not and #208?

**Comment by robertoostenveld** [comment]
I have just deleted the premature tweet and reset access permissions to prevent this from happening again, since apparently there is still no agreement how it is supposed to work.

The strategy I want is that _news_ is not about only tweeting, but about reaching out to all our users. And it might involve _all_ our channels. The procedure I have in mind is

1. a news item gets written, either by hand, or automatically (following a template)
2. the post is sent to this repository (fieldtrip/website) as pull request
3. it gets reviewed, in case it is not so news-worthy, the reviewer rejects the PR (closes it without merge) 
4. when it passes the initial review, it gets curated 
    - manual improvements are made to the longer content (body) and short tweet (header)
    - the PR gets merged
5. the tweet gets sent automatically (by a webhook)

If a news item such as "Hi everyone, we have just updated our communication strategy and from now on we will tweet less" is deemed news-worthy by someone, it enters at step 1 and goes all the way to 5 (or gets kicked out at 3). When merged, the tweet gets sent and the news item also appears on the website for a more persistent appearance and to reach a bigger and more inclusive audience. In this particular example, the short tweet is ... well short, like it was. The longer message to appear on the frontpage should inform people about the news as well, but also direct people to our twitter account, and should point to online documentation on the procedure (like the explanation here) so that the next generation people that care know how it works.  

Using this repository in the process also means we keep track of who suggests news items, who reviews them, who curates them, and who merges them (and triggers the news item to appear on the website and as tweet). Oh, and that could all be the same person.

**Comment by robertoostenveld** [comment]
>  I imagine you meant #280 not and #208?

correct

**Comment by elshafeh** [comment]
I apologise for probably missing my cues here and overstepping the five steps I was not aware of. It is ironic though for miscommunication to occur within an attempt to improve communication. I do not think that all channels should be conveying the same messages i.e. it doesn't make sense to me that announcing a change in twitter communication strategy should pop up in the news section. And while I did not appreciate the communication gap (and the subsequent reset of permissions etc.), I’m all ears and eyes to see what this develops into :)

**Comment by robertoostenveld** [comment]
if we want to have more people following us on twitter because we have improved the content of the tweets, it makes sense IMHO to announce that on a location that is _not_ twitter. Saying on twitter "hey we have (or will) changed" is rather superfluous, people that are already following http://twitter.com/fieldtriptoolbx will notice it there anyway.

**Comment by robertoostenveld** [comment]
Not all channels convey the same information or in the same format: Note the difference in the example posts that I made in the PRs between heading (tweet) and body (news item). But all channels are catered through the same mechanism, to ensure that on one channel we don't forget that the other exists as well. E.g. you might forget that there is news to be communicated on the website, or on the mailing list, whereas I might forget that there is news to be communicated on twitter. 

We could extend the idea even further of what information (be it news or not) appears where and how. E.g. by posting on other channels (fb, slack, discord) or by formatting existing channels differently. For example on the website front page we could make a **latest release** section that stands out from the **news and announcements** section.

**Comment by elshafeh** [comment]
> if we want to have more people following us on twitter because we have improved the content of the tweets, it makes sense IMHO to announce that on a location that is _not_ twitter. Saying on twitter "hey we have (or will) changed" is rather superfluous, people that are already following http://twitter.com/fieldtriptoolbx will notice it there anyway.

It is not superfluous. The ratio between the number of followers and the impact our tweets do not add up i.e. the account has been muted or whatever. IMHO that audience we're reaching out only through a mailing list have already made up their minds up on twitter and thus not to be targeted. And in any case, if that announcement has been deemed news-y, it could have turned into a news item later on, instead of deleting the tweet and denying me access. I still believe in a relative independence of communication channels :)

**Comment by robertoostenveld** [comment]
let's try to get some content news out...

**Comment by elshafeh** [comment]
> let's try to get some content news out...

Alrighty. Forward us the new credentials and I’ll work on it this afternoon. Don’t know if @Spaak wants to add something?

**Comment by Spaak** [comment]
> 5. the tweet gets sent automatically (by a webhook)

Is point (5) already how it works now (i.e., once we merge the PRs there will be tweets), or is this still to-be-implemented? I can still see auto-generated commit tweets in our Twitter account; those will have to be stopped before we enable the new functionality.

I agree with Hesham that announcing the new approach (specifically: we've stopped tweeting all commits and aim to increase our SNR) on Twitter itself is very relevant. Different users (e.g. us) would retweet the tweet, (possibly) causing our account to pop up again in all those people's feeds who have muted us.

**Comment by robertoostenveld** [comment]
Yes, it is implemented but not fully tested.

The automatic handling of point 1 and 2 are implemented with https://github.com/fieldtrip/automation/blob/master/release-newsitem.sh. This is called from a fieldtrip/fieldtrip webhook. I used a manual trigger of this hook to generate 277-280.

Point 5 is impemented with https://github.com/fieldtrip/automation/blob/master/send-tweet-newsitem.sh. This is called from a fieldtrip/website webhook, i.e. the post being merged causes the tweet to be sent and the website to be rebuilt.

**Comment by Spaak** [comment]
I did #277 (closed) and #278 (made some edits but not merged yet--see there).

**Comment by robertoostenveld** [comment]
so #279 and #280 somehow have the previous/current/next version messed up. I propose I delete those branches and try to generate them once more. Agreed?

**Comment by robertoostenveld** [comment]
#278 has the link to github `previous to current` correct, so as far as I can tell that is ready to merge. Please be my guest, and let's see whether the tweet works.

**Comment by Spaak** [comment]
@robertoostenveld I'd say we want to stop the automatic tweeting at the same time as (or before) establishing the new tweeting scheme, would you agree?

**Comment by robertoostenveld** [comment]
No, let’s first see if the new tweets work, then we switch off the old.

 There is no urgency to stop doing what we have been doing for 10 years

Verstuurd vanaf mijn iPhone

**Comment by Spaak** [comment]
> There is no urgency to stop doing what we have been doing for 10 years

Spoken like a true conservative ;)

But of course, I agree, there is no urgency here whatsoever. Just as long as we realize that any tweets we send out now will probably not get noticed by many people. This is perfectly fine for testing of course, but we'll probably want to save any really important new-style high-quality tweets until the auto-commit-tweets have stopped. On a related note: we might want to have at least a few days pause in between the last old-style tweet and the first new one, to increase the chances of the new one being seen.

**Comment by elshafeh** [comment]
I agree with @Spaak on this one. This will quickly get messy. Of course, it's important to try out all sorts of scripts but I strongly believe that we should be spending as much effort on rendering the tweets (and the account behind it) less impersonal and more engaging. Same goes for any public outreach.

**Comment by robertoostenveld** [comment]
I fully agree that we should spend our time on effective public outreach. That is also why I had asked you to merge and tweet last week (which did not happen, hence I am taking action myself). What do you think **you** can do about this?

**Comment by elshafeh** [comment]
Good question. I think it was clear what **I** can (and did) about this with my actions being labelled premature, access denied, and coming in terms with a plan in which I do not fully understand my role in, nor entirely agree to :) 

Is it simply clicking on the merge button? Why not, but this not what I had in mind for a contribution. Is it editing the tweets (& news items) before they are sent out? Sure, but I do not agree that this should happen unannounced or while the automated commits are on & I do not feel like we agree upon a strategy here nor that opinions are considered taken into account ... 

I believe we all do want to contribute something meaningful here, so let's do that :)

**Comment by robertoostenveld** [comment]
I am happy with how the releases and news items are now being announced. I hope you as well.
